WordPress/src/main/java/org/wordpress/android/ui/stats/StatsActivity.java

```diff
@@ -376,7 +386,40 @@ public void onRefreshRequested(StatsService.StatsEndpointsEnum[] endPointsNeedUp
         if (mIsUpdatingStats) {
             return;
         }
-        refreshStats(mCurrentTimeframe, mRequestedDate, true);
+
+        // fragments that needs an update require it at almost the same time. Keep a list of fragments
+        // that require the update and call the service with the right parameters.
+        synchronized (fragmentsRefreshListSynchObj) {
```

Comment:
This routine is confusing me. Is the following correct?

* It's called by every fragment when an update is requested
* It adds to a list of endpoints that need to be updated
* The only purpose of that list is to determine whether graph data (visits) needs updating
* 500ms after it's called it does the actual refresh
* The reason for the 500ms delay is to give enough time for all the fragments to have called this routine before the refresh is started

If this is correct, then this routine honestly seems very hackish and overly-complex for its purpose. I don't know the code well enough to suggest an alternative, but is there not a simpler way to refresh that doesn't require the synchronized calls, list of endpoints, and delayed execution?